src/cargo/util/config.rs

```diff
@@ -659,18 +661,25 @@ impl Config {
     fn load_credentials(&self, cfg: &mut ConfigValue) -> CargoResult<()> {
         let home_path = self.home_path.clone().into_path_unlocked();
         let credentials = home_path.join("credentials");
-        if fs::metadata(&credentials).is_err() {
+        let metadata = fs::metadata(&credentials);
+        if metadata.is_err() {
             return Ok(());
         }
-
+        let metadata = metadata?;
         let mut contents = String::new();
-        let mut file = File::open(&credentials)?;
-        file.read_to_string(&mut contents).chain_err(|| {
-            format!(
-                "failed to read configuration file `{}`",
-                credentials.display()
-            )
-        })?;
+
+        if cfg!(not(unix)) || (metadata.mode() & 0100 == 0) {
```

Comment:
I (surprisingly) couldn't find something better to do than `metadata.mode() & 0100 == 0` to check if the file is executable.
This is technically incorrect, as the executable bit cargo should check depends on the file's ownership.